src/arc-final.md

```diff
@@ -0,0 +1,80 @@
+# Final Code
+
+Here's the final code, with some added comments and re-ordered imports:
+```rust
+use std::marker::PhantomData;
+use std::ops::Deref;
+use std::ops::Drop;
+use std::ptr::NonNull;
+use std::sync::atomic::{self, AtomicUsize, Ordering};
+
+pub struct Arc<T> {
+    ptr: NonNull<ArcInner<T>>,
+    _marker: PhantomData<ArcInner<T>>,
+}
+
+pub struct ArcInner<T> {
+    rc: AtomicUsize,
+    data: T,
+}
+
+impl<T> Arc<T> {
+    pub fn new(data: T) -> Arc<T> {
+        // We start the reference count at 1, as that first reference is the
+        // current pointer.
+        let boxed = Box::new(ArcInner {
+            rc: AtomicUsize::new(0),
```

Comment:
shouldn't this be 1 instead of 0?

Reply:
Thanks for catching that, I'm dumb :thinking: